### PR TITLE
Sign a parent in when they accept, instead of mailing a second link

### PR DESCRIPTION
**Pull request by BRCornelius**
## Summary

Accepting an invitation landed the parent on a page whose only control was "Send me a sign-in link" — a second email to find, in a flow that started with an email. Parents stopped there. Accepting now establishes the session itself, so onboarding is one email and one click.

Reported during #9's real-parent validation weekend: "parents who accept the sign up link end up on a page where they can request a link."

## Key Decisions

**The token already proves what the second email proved.** The invitation token is mailed to the invited address and nowhere else, which is exactly the proof of mailbox possession a magic link gives. The second round trip bought no security and cost the app every parent who did not go back to their inbox.

This revises design-doc.md #4 Decision 1, which chose the two-step flow to keep `acceptInvitations` the single place memberships are granted. That reasoning still holds and is preserved: the accept action calls the same function `events.signIn` calls, so there is still one grant path. Only the source of the session changes.

**Writing the session row directly, rather than reaching for Auth.js.** Auth.js v5 exposes no "sign this user in" API under the database strategy — `signIn` for an email provider always mails a link and returns nothing usable, and the Credentials provider forces JWT sessions, which this app does not use. Two alternatives were considered and rejected:

- *Mint a `VerificationToken` and redirect into Auth.js's own callback.* Requires reimplementing `@auth/core`'s `hashToken` (sha256 of token + secret), an internal detail with no stable import. Gets it wrong silently.
- *Keep the magic link but suppress the email.* Needs the emitted URL captured out of `sendVerificationRequest`, which means AsyncLocalStorage plumbing around a provider instance that exists only to be intercepted.

Instead `src/lib/sessions.ts` inserts the `Session` row the way `@auth/core` does — `crypto.randomUUID()` token, `expires` at now + max age — and the action sets the cookie. Verified against `@auth/core@0.41.3` (`lib/actions/session.js`) and `@auth/prisma-adapter` that `auth()` needs nothing beyond that row and a matching cookie: no `Account`, no `VerificationToken`. Database sessions hand out an opaque row key rather than an encrypted payload, which is what makes writing one from outside Auth.js viable at all.

**One definition of the session cookie.** `proxy.ts` already had the two cookie names inlined, so this change would have made three places that must agree. `src/lib/session-cookie.ts` restates Auth.js's defaults once and all three import it. `usesSecureCookies` mirrors `createActionURL`'s protocol resolution (`AUTH_URL` → `x-forwarded-proto` → https) — disagreeing there does not fail subtly, since a `__Secure-` cookie sent over plain HTTP is dropped by the browser outright.

**Accepting stays a POST.** Making the emailed link itself sign the parent in was considered and rejected: corporate mail scanners follow every link in a message before the recipient opens it. On GET, the scanner would consume the invitation and the parent would arrive to "Already accepted" with no way in. The page render stays side-effect free; the POST is what consumes.

**The grant runs before the session write.** If the session insert fails, the parent is already a member and `/signin` admits them. The reverse would sign them in with no team to land on.

## What's in Scope

- `src/lib/sessions.ts` — new. `createUserSession`, plus the 90-day session lifetime constants, which `src/auth.ts` now imports rather than declaring its own copy.
- `src/lib/session-cookie.ts` — new, pure. Cookie name, secure-cookie resolution, and cookie attributes. `proxy.ts` now imports the names it used to inline.
- `resolveInvitedUser` in `src/lib/invitations.ts` — an invited coach has no `User` row yet (only guardians get one, from `linkGuardian`), and `emailVerified` is stamped at the point `@auth/core` stamps it. Case-insensitive lookup, matching the rest of the module.
- `/invite/[token]` — `requestInvitationLinkAction` becomes `acceptInvitationAction`: grants the memberships, writes the session, redirects to `/t/[teamId]`. The page's copy and its `?sent=1` state become an accept button and an `?error=1` state.
- Docs: the onboarding bullet in `product-brief.md` (a recorded decision genuinely revised, not just code drift) and a gotcha in `AGENTS.md` about the cookie now having two writers.

### Out of Scope

- The magic link is untouched. `/signin` still sends one, and it remains how a parent whose session lapses next season gets back in — admitted by the `Membership` this action created.
- The archived `#4` planning doc is left as it was. The revision is recorded in the action's own comment instead.
- No schema change, no migration.

## Known Gaps / Follow-ups

- **No database was available in this session, so the cookie write has not been exercised end to end against live Postgres.** The row shape and cookie contract were verified by reading `@auth/core` and the Prisma adapter, not by signing in. If a parent accepts and lands back on `/signin`, the cookie name or `secure` flag is the first place to look.
- `usesSecureCookies` falls back to `true` when neither `AUTH_URL` nor `x-forwarded-proto` is present, faithfully matching `createActionURL`. Next sets the header, so this should not arise, but a local setup that strips it would write a `__Secure-` cookie over HTTP and the browser would drop it.

## Test Plan

- [ ] `pnpm check` (lint → typecheck → 764 tests) and `pnpm exec next build`. Both green on this branch; `next build` rather than `pnpm build` because the latter runs `prisma migrate deploy` first and needs `DATABASE_URL`.
- [ ] Invite a parent from `/t/[teamId]/roster/[entryId]`, open the invitation email, click **Accept invitation**, then click **Accept invitation** on the page. Expected: land on `/t/[teamId]` already signed in, with no second email involved. Confirm exactly one email arrived.
- [ ] Confirm the session actually persists — reload `/t/[teamId]` and navigate to another team page; Proxy must not bounce you to `/signin`. In devtools the cookie should be `__Secure-authjs.session-token` in production, `authjs.session-token` over local HTTP.
- [ ] Reopen the same invitation link. Expected: "Already accepted", with a working link to `/signin` — and signing in there still works, since the parent now holds a `Membership`.
- [ ] Invite a **coach** from `/t/[teamId]/members` to an address with no existing `User` row, and accept. Expected: same result, arriving with the COACH role. This is the path `resolveInvitedUser` exists for.
- [ ] Edge cases checked in unit tests rather than by hand: unknown token, expired token, already-accepted token, grant failure, and session-write failure after a successful grant.

---
_Generated by [Claude Code](https://claude.ai/code/session_014wXsgRvpaqf3dNkVBFQych)_

### CODE REVIEW

File: src/app/invite/[token]/actions.ts [hunk 1]
```diff
@@ -1,51 +1,95 @@
...
+  // parent on the invite page with an error rather than at a team URL they
+  // have no session for.
+  let destination = `/invite/${token}?error=1`;
+
   try {
     const invitation = await getInvitationByToken(token);
 
+    // Re-checked rather than trusted from the page render: seconds pass
+    // between the page loading and this submitting, and the invitation can
+    // expire or be revoked in between.
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/app/invite/[token]/actions.ts:66]: Fixed in b631d85. A single `now` is captured at the top of the action and passed to `isLiveInvitation`, `resolveInvitedUser`, `acceptInvitations`, and `createUserSession`. All four already took `now` as a parameter so the caller could pin it — the action just wasn't passing one.

Worth noting the blast radius you identified is narrower than it looks but real: a parent's `Membership` is created by `linkGuardian` when they're first linked, so they'd land fine regardless. It's an invited **coach**, whose membership only exists after acceptance, who would get a session and then be rejected by `requireTeamAccess` on the team page.

Added a regression test that advances the fake clock past the invitation's expiry inside the `resolveInvitedUser` call, forcing exactly that interleaving; it fails against the pre-fix action and passes now.

One part of the scenario I deliberately left open, and documented in the action rather than half-guarding: a shared `now` closes the expiry race but not the read-then-write gap, so a coach revoking the invitation between `getInvitationByToken` and `acceptInvitations` still consumes zero rows. Bailing on a zero count would be the wrong guard — because a parent's `Membership` predates the invitation, revoking it leaves them genuinely entitled to the team, so a count check would lock out the common case to catch a millisecond of the rare one. Doing it correctly means "zero consumed **and** no membership on this team", an extra query for a millisecond window, which didn't seem worth it.

---
_Generated by [Claude Code](https://claude.ai/code)_
